reference-implementation/to-upstream-wpts/transform-stream/flush.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+if (self.importScripts) {
+  self.importScripts('/resources/testharness.js');
+}
+
+promise_test(() => {
+  let flushCalled = false;
+  const ts = new TransformStream({
+    transform() { },
+    flush() {
+      flushCalled = true;
+    }
+  });
+
+  return ts.writable.getWriter().close().then(() => {
+    return assert_true(flushCalled, 'closing the writable triggers the transform flush immediately');
+  });
+}, 'TransformStream flush is called immediately when the writable is closed, if no writes are queued');
+
+// TODO
+test(() => {
+  let flushCalled = false;
+  const ts = new TransformStream({
+    transform() {
+      return new Promise(resolve => setTimeout(resolve, 10));
+    },
+    flush() {
+      flushCalled = true;
+      return new Promise(); // never resolves
```

Comment:
This is not valid promise usage; it should be `new Promise(() => {})`. A preexisting problem I guess!